packages/steveo/src/runtime/workflow.ts

```diff
@@ -181,62 +212,85 @@ export class Workflow {
+          workflowName: this.name,
+        });
 
-      // Initialise new workflow execution?
-      if (!payload.workflowId) {
-        const firstStep = this.steps[0];
+        // Initialise new workflow execution?
+        if (!payload.workflowId) {
+          const firstStep = this.steps[0];
 
-        this.logger.debug(`No workflow in payload, initialising new workflow`);
+          logger.debug({
+            message: `No workflow in payload, initialising new workflow`,
+          });
 
-        await repos.workflow.workflowInit({
-          workflowId,
-          serviceId: this.options.serviceId,
-          current: firstStep.name,
-          initial: payload,
-        });
-      }
+          await repos.workflow.workflowInit({
+            workflowId,
+            serviceId: this.options.serviceId,
+            current: firstStep.name,
+            initial: payload,
+          });
+        }
 
-      const state = await this.loadState(workflowId, repos);
+        const state = await this.loadState(workflowId, repos);
 
-      if (!state.current) {
-        throw new Error(`Workflow ${workflowId} step was undefined`);
-      }
+        if (!state.current) {
+          throw new AppError(this.logger, {
+            message: `Workflow state was not found`,
+            workflowId,
+          });
+        }
 
-      const step = this.steps.find(s => s.name === state.current);
-      if (!step) {
-        throw new Error(
-          `Worflow ${workflowId} could not find step ${state.current}`
-        );
-      }
+        const step = this.steps.find(s => s.name === state.current);
+        if (!step) {
+          throw new AppError(this.logger, {
+            message: `Worflow could not find current step`,
+            workflowId,
+            current: state.current,
+          });
+        }
 
-      this.logger.debug(
-        `Load state for workflow ${workflowId}, step ${step.name}`
-      );
+        logger.debug({
+          message: `Load state for workflow step`,
+          stepName: step.name,
+        });
 
-      const context: ExecuteContext = {
-        workflowId,
-        payload,
-        state,
-        step,
-        repos,
-      };
-
-      // If this step has failed then we pass control over to the rollback executor.
-      if (state.errors?.length) {
-        await this.executeRollback(context);
-        return;
-      }
+        const context: ExecuteContext = {
+          workflowId,
+          payload,
+          state,
+          step,
+          repos,
+        };
+
+        // If this step has failed then we pass control over to the rollback executor.
+        if (state.errors?.length) {
+          await this.executeRollback(context);
+        } else {
+          await this.executeNextStep(context);
+        }
 
-      await this.executeNextStep(context);
-    });
+        return workflowId;
+      });
+    } catch (err) {
+      this.logger.error({
+        message: 'Subscribe processor error',
```

Comment:
Sanity checking that we don't need to emit an event here?

It might be rolling up somewhere but hard to wrap my head around in context of a pull request.

Reply:
Yeah good catch, I had a little revise of the events emitted as they all needed a little tweak because they talked in terms of the "workflow", when they were actually only working on a "workflow step"